UnitTests/UnitTests.HighPerformance.Shared/Extensions/Test_ReadOnlySpanExtensions.cs

```diff
@@ -0,0 +1,130 @@
+using Microsoft.Toolkit.HighPerformance.Extensions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace UnitTests.HighPerformance.Extensions
+{
+    [TestClass]
+    public partial class Test_ReadOnlySpanExtensions
+    {
+        [TestCategory("ReadOnlySpanExtensions")]
+        [TestMethod]
+        public void Test_ReadOnlySpanExtensions_DangerousGetReference()
+        {
+            ReadOnlySpan<int> data = CreateRandomData<int>(12, default).AsSpan();
+
+            ref int r0 = ref data.DangerousGetReference();
+            ref int r1 = ref Unsafe.AsRef(data[0]);
+
+            Assert.IsTrue(Unsafe.AreSame(ref r0, ref r1));
+        }
+
+        [TestCategory("ReadOnlySpanExtensions")]
+        [TestMethod]
+        public void Test_ReadOnlySpanExtensions_DangerousGetReferenceAt_Zero()
+        {
+            ReadOnlySpan<int> data = CreateRandomData<int>(12, default).AsSpan();
+
+            ref int r0 = ref data.DangerousGetReference();
+            ref int r1 = ref data.DangerousGetReferenceAt(0);
+
+            Assert.IsTrue(Unsafe.AreSame(ref r0, ref r1));
+        }
+
+        [TestCategory("ReadOnlySpanExtensions")]
+        [TestMethod]
+        public void Test_ReadOnlySpanExtensions_DangerousGetReferenceAt_Index()
+        {
+            ReadOnlySpan<int> data = CreateRandomData<int>(12, default).AsSpan();
+
+            ref int r0 = ref data.DangerousGetReferenceAt(5);
+            ref int r1 = ref Unsafe.AsRef(data[5]);
+
+            Assert.IsTrue(Unsafe.AreSame(ref r0, ref r1));
+        }
+
+        [TestCategory("ReadOnlySpanExtensions")]
+        [TestMethod]
+        [SuppressMessage("StyleCop.CSharp.SpacingRules", "SA1009", Justification = "List<T> of value tuple")]
+        public void Test_ReadOnlySpanExtensions_Enumerate()
+        {
+            ReadOnlySpan<int> data = CreateRandomData<int>(12, default).AsSpan();
+
+            List<(int Index, int Value)> values = new List<(int, int)>();
+
+            foreach (var item in data.Enumerate())
+            {
+                values.Add(item);
+            }
+
+            Assert.AreEqual(values.Count, data.Length);
+
+            for (int i = 0; i < data.Length; i++)
+            {
+                Assert.AreEqual(data[i], values[i].Value);
+                Assert.AreEqual(i, values[i].Index);
+            }
+        }
+
+        [TestCategory("ReadOnlySpanExtensions")]
+        [TestMethod]
+        public void Test_ReadOnlySpanExtensions_Tokenize_Empty()
+        {
+            string text = string.Empty;
+
+            var result = new List<string>();
+
+            foreach (var token in text.AsSpan().Tokenize(','))
+            {
+                result.Add(new string(token.ToArray()));
+            }
+
+            var tokens = text.Split(',');
+
+            CollectionAssert.AreEqual(result, tokens);
+        }
+
+        [TestCategory("ReadOnlySpanExtensions")]
+        [TestMethod]
+        public void Test_ReadOnlySpanExtensions_Tokenize_Csv()
```

Comment:
Might want to change these names later to `Comma` as CSV has other parsing implications 😋

Reply:
Ahahahah right 😄
Done in https://github.com/windows-toolkit/WindowsCommunityToolkit/pull/3273/commits/6f488cd372a2b97c82e903bfe86b9732fef9fc27.

Reply:
nice information Admin...
i suggest if anyone want full version of latest software he should visit [this site](https://propercracks.com/)